pkg/patterns/declarative/reconciler.go

```diff
@@ -166,53 +191,68 @@ func (r *Reconciler) BuildDeploymentObjects(ctx context.Context, name types.Name
 	log := log.Log
 
 	// 1. Load the manifest
-	manifestStr, err := r.loadRawManifest(ctx, instance)
+	manifestFiles, err := r.loadRawManifest(ctx, instance)
 	if err != nil {
 		log.Error(err, "error loading raw manifest")
 		return nil, err
 	}
-
+	mainfestObjects := &manifest.Objects{}
 	// 2. Perform raw string operations
-	for _, t := range r.options.rawManifestOperations {
-		transformed, err := t(ctx, instance, manifestStr)
+	for manifestPath, manifestStr := range manifestFiles {
+		for _, t := range r.options.rawManifestOperations {
+			transformed, err := t(ctx, instance, manifestStr)
+			if err != nil {
+				log.Error(err, "error performing raw manifest operations")
+				return nil, err
+			}
+			manifestStr = transformed
+		}
+
+		// 3. Parse manifest into objects
+		objects, err := manifest.ParseObjects(ctx, manifestStr)
 		if err != nil {
-			log.Error(err, "error performing raw manifest operations")
+			log.Error(err, "error parsing manifest")
 			return nil, err
 		}
-		manifestStr = transformed
-	}
 
-	// 3. Parse manifest into objects
-	objects, err := manifest.ParseObjects(ctx, manifestStr)
-	if err != nil {
-		log.Error(err, "error parsing manifest")
-		return nil, err
-	}
+		// 4. Perform object transformations
+		transforms := r.options.objectTransformations
+		if r.options.labelMaker != nil {
+			transforms = append(transforms, AddLabels(r.options.labelMaker(ctx, instance)))
+		}
+		// TODO(jrjohnson): apply namespace here
+		for _, t := range transforms {
+			err := t(ctx, instance, objects)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-	// 4. Perform object transformations
-	transforms := r.options.objectTransformations
-	if r.options.labelMaker != nil {
-		transforms = append(transforms, AddLabels(r.options.labelMaker(ctx, instance)))
-	}
-	// TODO(jrjohnson): apply namespace here
-	for _, t := range transforms {
-		err := t(ctx, instance, objects)
-		if err != nil {
-			return nil, err
+		// 5. Write objects to filesystem for kustomizing
+		for _, item := range objects.Items {
+			json, err := item.JSON()
+			if err != nil {
+				log.Error(err, "error convert object to json")
+				return nil, err
+			}
+			_fSys.WriteFile(string(manifestPath), json)
 		}
+		mainfestObjects.Items = append(mainfestObjects.Items, objects.Items...)
 	}
+	// 6. Sort objects to work around dependent objects in the same manifest (eg: service-account, deployment)
+	mainfestObjects.Sort(DefaultObjectOrder(ctx))
 
-	// 5. Sort objects to work around dependent objects in the same manifest (eg: service-account, deployment)
-	objects.Sort(DefaultObjectOrder(ctx))
+	// 6. Sort objects to work around dependent objects in the same manifest (eg: service-account, deployment)
```

Comment:
Why calls `mainfestObjects.Sort` twice? If you have a reason for calling this, please add a comment or it is a just typo?

Reply:
It's just typo. Thanks. :-)